src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -148,35 +158,106 @@ public void visitJavadocToken(DetailNode ast) {
+                    && getContentOfChild(child).equals(INHERIT_DOC)) {
+                result.append(INHERIT_DOC);
             }
-            else if (!SKIP_TOKENS.contains(child.getType())) {
-                extraTextFound = true;
+            else if (child.getType() == JavadocTokenTypes.HTML_ELEMENT
+                    && CommonUtils.isBlank(result.toString().trim())) {
+                result.append(getStringInsideTag(result.toString(),
+                        child.getChildren()[0].getChildren()[0]));
             }
-            if (extraTextFound) {
+            else if (child.getType() == JavadocTokenTypes.JAVADOC_TAG) {
+                flag = false;
+            }
+            else if (isPeriodAtEnd(result.toString().trim(), child)) {
+                periodFound = true;
+            }
+            if (!flag) {
                 break;
             }
         }
-        return containsInheritDoc && !extraTextFound;
+        return validate(ast, periodFound, result.toString());
+    }
+
+    /**
+     * Returns content when token type is javadoc inline tag.
+     * @param child javadoc inline tag ast.
+     * @return content of child nodes as string.
+     */
+    private static String getContentOfChild(DetailNode child) {
+        final StringBuilder contents = new StringBuilder();
+        for (DetailNode node : child.getChildren()) {
+            contents.append(node.getText().trim());
+        }
+        return contents.toString();
+    }
+
+    /**
+     * Concatenates string within text of html tags.
+     * @param result javadoc string
+     * @param detailNode javadoc tag node
+     * @return java doc tag content appended in result
+     */
+    private String getStringInsideTag(String result, DetailNode detailNode) {
+        final StringBuilder contents = new StringBuilder(result);
+        DetailNode tempNode = detailNode;
+        while (tempNode != null) {
+            if (tempNode.getType() == JavadocTokenTypes.TEXT) {
+                contents.append(tempNode.getText());
+            }
+            tempNode = JavadocUtils.getNextSibling(tempNode);
+        }
+        return contents.toString();
+    }
+
+    /**
+     * Checks if period is set at end of first sentence.
+     * @param result javadoc string
+     * @param child child node
+     * @return true if period is at end of first sentence.
+     */
+    private boolean isPeriodAtEnd(String result, DetailNode child) {
+        return child.getType() == JavadocTokenTypes.NEWLINE
+                && !CommonUtils.isBlank(result)
+                && String.valueOf(result
+                    .charAt(result.length() - 1)).equals(period);
+    }
+
+    /**
+     * Finds if javadoc ends with period or not.
+     * @param ast ast Javadoc root node.
+     * @param periodFound true if period is found in sentence before end of line
+     * @param result javadoc string
+     * @return error string
+     */
+    private String validate(DetailNode ast,
```

Comment:
please return string to `visitJavadocToken` method and do all validation of String that is first sentence there.